### DISCUSSION TITLE
Some observations about the Friesach organ

### DISCUSSION THREAD

**Discussion by ClarkeBrunt** [GrandOrgue/grandorgue#1407] [General]
Hi all.

I recently acquired a midi console, so can finally use GO properly - for a year or two, I've been following it's development with interest, but not actually using it.

So far, I've only used the demo organ and Piotr Grabowski's Friesach organ. I'm a retired programmer, so can't resist looking in the ODF, and wondering how things work.

Anyway, a few observations - maybe well known to experienced folk, but perhaps not to others.

The Friesach organ fits in 16GB RAM without missing anything out, or reducing the sample size below 24 bits. It uses maybe 12-13GB - this is with lossless compression enabled (and on Windows 10, should it matter).

Computer is a Dell Optiplex 7010 (small form factor), an i3-3240 @ 3.40 GHz, which is I guess 'nothing special' by modern standards, yet I see no sign so far of being able to overload it, even with lots of stops and notes. Sound and midi are via a Behringer u-phoria umc404hd connected to USB.

I hadn't expected all the sound effects - blower, stop noises, etc. Certainly makes for a 'realistic' experience!

The Friesach organ seems rather quiet at GO's default -15dB. On Piotr's website (the page about GrandOrgue versions of his sample sets), he says "volume is normalized to the default GrandOrgue setting, which is -15dB". I sort of wonder if something has been allowed for twice, anyway volume can always be turned up in GO. Are his other sets quiet as well?

While looking in the ODF, I noticed that there is a visible control on the main panel for switching the blower on and off. It's so small that I hadn't noticed it - black button above top-right of 3rd manual. It really does turn the blower off - not just the noise! The organ stops making sound, and there is a lengthy sound of bellows emptying etc. Press again to turn it back on. I don't think any of the other tiny 'controls' shown on the main panel actually do anything.

There seems to be no visible General Cancel on the organ, but the one in GO's 'Combination Setter' panel can be used (and programmed for midi). Is that the only GC button which GO supplies - I looked in the other panels?

I could see in the ODF that there were supposed to be key and pedal action noises, but I wasn't hearing any. I believe I've tracked down why. E.g. in [Stop100] Name=Key action Pedal Attack (and equivalent sections for Release, and for the 3 manuals), GCState isn't specified. This controls how the stop responds to General Cancel. GO help (for Drawstop) says "(integer -1 to 1, default: implementation defined) State of the button after pressing GC. -1 means no change, 0 off and 1 on.". In this case 'implementation defined' seems to mean that these stops were turned off by General Cancel, which inevitably I'd pressed before saving settings. There is no visible control to turn them on again. Presumably could be fixed by inserting GCState=-1.

As I say, there is no visible control for these 'noise stops', but they can be turned on/off by toggling them in the 'MIDI Objects' list (menu bar Audio/Midi) - use 'Status' to check whether on or off. They could also be programmed to respond to Midi here.

The noises have their own 'Windchest' [WindchestGroup005] Name=Noises, so the volume of some noises can be controlled by the 'Noises' expression pedal in GO's 'Coupler manuals and volume' panel. For some reason (error by Piotr maybe?) the pedal and key action noises use [WindchestGroup004]
Name=4. Solowerk, so are controlled (along with the Solowerk) by that expression pedal.

That's enough for now. I wonder what other interesting 'features' I'll find!

**Comment by jason-barba** [comment]
@ClarkeBrunt
[GO_OFD_architecture.pdf](https://github.com/GrandOrgue/grandorgue/files/10825953/GO_OFD_architecture.pdf)
: I wrote this, mostly for my own understanding of ODF. Enjoy!

**Comment by ClarkeBrunt** [comment]
Further to my comment earlier, I've now 'fixed' the Friesach organ by editing the ODF so that (a) key action noises aren't turned off by General Cancel, and (b) the key action noises are allocated to the Noises windchest (rather than Solowerk).

The changes are easily described: search for 'key action' which occurs in 8 'stops' for the Attack and Release noises of pedal and 3 manuals. In each of these 8 'stop' sections, add:

GCState=-1 ; don't turn off with GC

and change WindchestGroup=4 to WindchestGroup=5

I also appended something to 'OrganComments', so I can check it's my edited version in the Organ Properties dialog.

**Comment by jefferycrowley** [comment]
I am curious if there is a way to add the expression shutter noise to an ODF.   I have searched the entire GitHub site for Grand Orgue, but I only came across ancient posts that remarked that this is not possible at this time.    

This is the Internet master Google's AI answer - there is  much more than this, but it all involves an external step.  "_In short, while Grand Orgue doesn't have a built-in "shutter noise" feature, it's possible to incorporate that sound using various audio techniques, both within the software and through external integration_."

I have actual shutter noise sounds as does Piotr Grabowski's Freisach and St. John Cantius Samplesets.    This works in HW but not in GO.

Is it possible to link the *.wav file to a "Stage" of the Expression Pedal - i.e. when you deploy the expression pedal it plays a *.wav file (as pipe 001 or 007 etc.) all the way to fully open or closed.

I have tried to figure this out by reading the HW XML file and since the file is so large I cannot find the spot where the sound files for the expression pedals are - so, I cannot see how to set this up in GO

**Reply by eturpault** [reply]
If you open the HW ODF with OdfEdit, you can navigate in the sections of the ODF in a tree view way (in the tab "HW sections" at the right side of the application, use a double-click on a parent or child section to select it), which is more convenient than reading in the xml file.

The expression pedal is the section ContinuousControl000001 defined in the section DisplayPage000001. It controls several ContinuousControlStageSwitch sections and ContinuousControl... sections through ContinousControlLinkage or ContinousControlDoubleLinkage sections. By selecting the successive children of these sections across several sub-levels, you will reach ContinousControl sections or Switch sections controlling Pipe_SoundEngine01_Layer or Pipe_SoundEngine01 sections, under which are defined the expression pedal noise samples (with files names starting by "Noises/Encl_").

Another approach (down to top) is to search for the string "Noises/Encl_" in the HW ODF (tab "Search & replace" of OdfEdit, then press the button "HW search"). You will have 10 search results. Double clicking one result will show this section in the tab "HW sections", from which you can select the successive parents until reaching the controlling ContinousControl sections.

The noise samples are controlled through a tricky network of continous control sections. To try to understand it, one would need to make a relational diagram between the involved sections, and understand what is the behavior of each one.

**Reply by jefferycrowley** [reply]
Thanks- I have had ODF Edit for a couple of years, but have never used it!  I guess there is no time like the present!  I have used other software to read the XML documents, but they are just a murky with their different organization.   With most NOT reading Parent/Child either at all or very limited functionality but all of them you can search the entire file for such things.
